src/pages/DataCatalogue/components/DataCatalogueTable/DataCatalogueRow.tsx

```diff
@@ -0,0 +1,92 @@
+import { useEffect, useState } from 'react';
+
+import { Collection } from '@/typings/stac';
+
+type DataCatalogueRowProps = {
+  row: Collection;
+};
+
+type ParentData = {
+  id: string;
+  title: string;
+  links?: { rel: string; href: string }[];
+};
+
+export const DataCatalogueRow = ({ row }: DataCatalogueRowProps) => {
+  const [breadcrumb, setBreadcrumb] = useState<ParentData[]>([]);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    const fetchParentData = async (url: string): Promise<ParentData | null> => {
+      try {
+        const response = await fetch(url);
+        const parentData = await response.json();
+        return parentData;
+      } catch (error) {
+        console.error('Error fetching parent data:', error);
+        return null;
+      }
+    };
+
+    const buildBreadcrumbTrail = async () => {
+      const trail: ParentData[] = [];
+      let currentUrl = row.links?.find((link) => link.rel === 'parent')?.href;
+
+      while (currentUrl) {
```

Comment:
Just want to make sure I fully understand your comment here, so there is a possibility due to some human error that a `collection`, `notebook` or something else could have cyclical dependency, which is actually a bug in the data structure and should be fixed there, but to save the whole app collapsing, we should maybe try to guard against it. I'm not a fan of trying to anticipate bad data, garbage in, garbage out as they say, but yes, we should maybe try to do something. I'd probably first resort to `Suspense` so that only the Data Catalog Table broken and it didn't affect the whole map. I guess my concern with your suggestion, however I don't have a better idea, is the potential amount of previous parents and constantly going back and forward up/down the hierarchy checking for data integrity. 

I wonder if it would be better at data ingestion time for such checks to be done, so the harvested data is never in a bad state.

Again, I'll give it some thought, talk to Tom to see what, if any harvesting checks are done etc.

Reply:
I spoke with tom and if my understanding of your question is correct, that a sub-catalog can reference a parent catalog, then no, that cannot happen, it could be a copy, which is still a data bug most likely, but it cannot have cyclical dependencies. and get into an infinite loop.

Reply:
Yes, it shouldn't happen.